embeddings/data/hugging_face_data_loader.py

```diff
@@ -0,0 +1,13 @@
+import datasets
+
+from embeddings.data.data_loader import DataLoader
+from embeddings.data.dataset import Dataset
+from embeddings.data.hugging_face_dataset import HuggingFaceDataset
+
+
+class HuggingFaceDataLoader(DataLoader[str, datasets.DatasetDict]):
+    def load(self, dataset: Dataset[str]) -> datasets.DatasetDict:
+        if isinstance(dataset, HuggingFaceDataset):
+            return datasets.load_dataset(dataset.dataset, **dataset.load_dataset_kwargs)
+        else:
+            raise ValueError("This DataLoader should be used with HuggingFaceDataset only.")
```

Comment:
mozemy zmienic ze nie dataloader przyjmuje typ K ktory jest podtypem datasetu  a nie data set, wtedy mozna zrobic dataloader tylko do huginga i nie robic takich ifow, albo zrobic ze dataset hugingface zawiera jakis hugingface data config i go zwraca, wtedy tutaj oczekujemy nie str a ten hugingface data config

Reply:
nie możemy tak zrobić. argument może funkcji może być tylko ogólniejszym typem niż ten z klasy abstrakcyjnej. Liskov substitution principle: https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides

a co do configu, to miałoby być coś pokroju `if data_config.type == 'huggingface':`?

Reply:
tak to przez to mi chodzi zeby zmienic jak to jest zrobione w kontekscie tego jak definiowany jest dataloader

Reply:
nie za bardzo rozumiem szczerze mówiąc. to jest jakaś dobra praktyka czy wybiegające plany? bo w tym przypadku nie widzę takiej potrzeby jeśli tylko jeden dataset bedzie typu 'huggingface', nawet jesli będzie zapotrzebowanie na więcej to przecież można dziedziczyć